scripts/monorepo/bump-all-updated-packages/index.js

```diff
@@ -0,0 +1,186 @@
+      {
+        cwd: ROOT_LOCATION,
+      },
+    );
+    echo();
+
+    await inquirer
+      .prompt([
+        {
+          type: 'list',
+          name: 'shouldBumpPackage',
+          message: `Do you want to bump ${packageName}?`,
+          choices: ['Yes', 'No'],
+          filter: val => val === 'Yes',
+        },
+      ])
+      .then(({shouldBumpPackage}) => {
+        if (!shouldBumpPackage) {
+          echo(`Skipping bump for ${packageName}`);
+          return;
+        }
+
+        const updatedVersion = bumpPackageVersion(
+          packageAbsolutePath,
+          packageManifest,
+        );
+        echo(
+          `\u2705 Successfully bumped ${chalk.green(
+            packageName,
+          )} to ${chalk.green(updatedVersion)}`,
+        );
+      });
+  };
+
+const buildAllExecutors = () => {
+  const executors = [];
+
+  forEachPackage((...params) => {
+    executors.push(buildExecutor(...params));
+  });
+
+  return executors;
+};
+
+const checkIfThereIsSomethingToCommit = () => {
+  const {stdout: thereIsSomethingToCommit} = exec('git status --porcelain', {
+    cwd: ROOT_LOCATION,
+    silent: true,
+  });
+
+  return Boolean(thereIsSomethingToCommit);
+};
+
+const main = async () => {
+  if (checkIfThereIsSomethingToCommit()) {
+    echo(
+      chalk.red(
+        'Found uncommitted changes. Please commit or stash them before running this script',
+      ),
+    );
+    exit(1);
+  }
+
+  const executors = buildAllExecutors();
+  for (const executor of executors) {
+    await executor()
+      .catch(() => exit(1))
+      .then(() => echo());
+  }
+
+  await inquirer
+    .prompt([
+      {
+        type: 'list',
+        name: 'shouldRunYarn',
+        message: 'Do you want to run yarn?',
+        choices: ['Yes', 'No'],
+        filter: val => val === 'Yes',
+      },
+    ])
+    .then(({shouldRunYarn}) => {
+      if (!shouldRunYarn) {
+        echo(`Skipping ${chalk.dim('yarn install')}`);
+        return;
+      }
+
+      const {code} = exec('yarn', {cwd: ROOT_LOCATION});
+
+      if (code) {
+        echo(chalk.red('Failed to run yarn'));
+        exit(code);
+      }
+    })
+    .then(() => echo());
+
+  if (checkIfThereIsSomethingToCommit()) {
+    await inquirer
+      .prompt([
+        {
+          type: 'list',
+          name: 'shouldSubmitCommit',
+          message: 'Do you want to submit a commit with these changes?',
+          choices: ['Yes', 'No'],
+          filter: val => val === 'Yes',
+        },
+      ])
+      .then(({shouldSubmitCommit}) => {
+        if (!shouldSubmitCommit) {
+          echo('Not submitting a commit, but keeping all changes');
+          return;
+        }
+
+        // exec from shelljs currently does not support interactive input
+        // https://github.com/shelljs/shelljs/wiki/FAQ#running-interactive-programs-with-exec
+        execSync('git commit -a', {cwd: ROOT_LOCATION, stdio: 'inherit'});
```

Comment:
can we "force" the commit message to have a specific format/message and modify CI so that it will only do the npm publish when the commit message has that specific format?

ex. always have as commit message `[CI] bump monorepo packages` and have CI only do the logic for npm publish on commits with that message?

I would like to reduce the risk of someone inadvertently:
* changing a version of a package manually as part of some PRs (ex. imagine someone from the community doing a change in a subpackage, and because they don't know how things work they bump the version in the package.json; that would rely on a human reviewer to catch it and I'm not a fan of relying on humans 🤣)
* only changing it in a subset of the places this needs to work on

Reply:
I am not sure if we should stick to some specific commit message. I understand the purpose, to avoid a situation, where we merge commit with a version bump and automatically publish this to npm.

Proposed solution won't save us from this happening, because this commit could also have the same message. Of course, this is less likely to happen

Maybe we should stick to some other artifacts on where to publish a package? Maybe commit author username?

Reply:
I'm proposing the commit message one because it's a common strategy used among many OSS repos - if you have a better one I'm open ears; I'm not sure I understand how the  "commit author username" would help preventing the situation I mentioned above because the only way I can see it working is by manually maintaining an extra list of "authorized authors", and even in that case is a very wide filter which I don't think will be safe enough.

Reply:
> I'm proposing the commit message one because it's a common strategy used among many OSS repos - if you have a better one I'm open ears; I'm not sure I understand how the "commit author username" would help preventing the situation I mentioned above because the only way I can see it working is by manually maintaining an extra list of "authorized authors", and even in that case is a very wide filter which I don't think will be safe enough.

I've added such logic, now running this script produces a commit with specific message. Then we have a workflow on CircleCI, which expects to receive package version bump only inside commit with this specific message